scripts/ninja.js

```diff
@@ -115,7 +115,7 @@ var useEnv = false;
  * Note ocamldep.opt has built-in macro handling OCAML_VERSION
  */
 var getOcamldepFile = () => {
-  if (useEnv) {
+  if (useEnv || process.env.ESY === "true") {
```

Comment:
When we're using `esy` it's like we have ocaml in the environment

Reply:
Would you prefer to make this something a little more specific? `ESY_BS_SUPPORT`?

Reply:
We could change to that, we have this environment variable in `buildocaml.js` as well but we can change everywhere if we want to be more specific